### ISSUE TITLE
PRM loss 疑问

### ISSUE THREAD

**Issue by EthanChen1234** [OpenRLHF/OpenRLHF#510]
您好，在阅读代码时有个疑问。

https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/models/loss.py#L287
`
for i, token in enumerate(self.reward_token_ids):
    labels = torch.where(labels == token, i, labels)
`

对 self.reward_token_ids index=0 给标签0，index=1 给标签1.

https://github.com/OpenRLHF/OpenRLHF/blob/main/examples/scripts/train_prm_mistral.sh#L26
我们在设置训练参数的时，是不是 bad 放前面，good 放后面，更直观一些？
`
--reward_tokens - + 
`

**Comment by hijkzzz** [comment]
--reward_tokens +  - 更安全

**Comment by EthanChen1234** [comment]
> --reward_tokens + - 更安全（soft label必须这样使用）

大佬，没有想明白为啥 --reward_tokens + - 更安全。方便给个详细的解释么，谢谢了

**Comment by Wloner0809** [comment]
> 写代码的时候默认 正的放前面的 非 soft 无所谓

So did you mean that when we use hard label, '+' token gets label 0 and '-' token gets label 1? Is this reasonable?
